[RUNTIME] Adds Student Code Console Logging

brandonxxlee opened the pull request:
Overrides print for studentCode process and redirects to TCP socket to be sent to dawn.

Redirects BadThing stack trace to TCP socket to be sent to dawn.

Still maintains regular stdout print for testing purposes.

Note: This is branched from runtime/TCPIntegration so includes a lot of other commits, don't merge until other PR is merged.

bodiddles committed:
[RUNTIME] Adds functionality to reset dawn's ip in runtime

Allows for resetting ip that runtime sends to avoid hard coding ip's
that runtime will send to.

When Dawn connects, its current ip address is stored into state manager and eventually sent to relevant processes.

Changes process spawning order to only spawn the UDP send process when Dawn connects. TCP process will also be spawned with this logic.

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral into runtime/re_ip

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral into runtime/re_ip

brandonxxlee committed:
Merge branch 'master' of github.com:pioneers/PieCentral into runtime/re_ip

bodiddles committed:
Merge branch 'runtime/re_ip' of https://github.com/pioneers/PieCentral into runtime/re_ip

brandonxxlee committed:
Merge branch 'runtime/re_ip' of github.com:pioneers/PieCentral into runtime/re_ip

brandonxxlee committed:
[RUNTIME] Lock IP on first connect to Dawn

brandonxxlee committed:
[RUNTIME] Start implementation of TCP

brandonxxlee committed:
[RUNTIME] Write main methods of TCP communication with Dawn

brandonxxlee committed:
[RUNTIME] Finish initial code, ready for testing

brandonxxlee committed:
[RUNTIME] Add method to terminate TCP if disconnected

brandonxxlee committed:
Merge branch 'master' of github.com:pioneers/PieCentral into runtime/TCPintegration

brandonxxlee committed:
[RUNTIME] Create ANSIBLE_COMMANDS enum class

brandonxxlee committed:
Merge branch 'master' of github.com:pioneers/PieCentral into runtime/TCPintegration

brandonxxlee committed:
[RUNTIME] Add Console Log command into runtimeUtil

bodiddles committed:
[RUNTIME] integrating tcp sockets

bodiddles committed:
Merge branch 'runtime/TCPintegration' of https://github.com/pioneers/PieCentral into runtime/TCPintegration

brandonxxlee committed:
[RUNTIME] Finish fake_dawn to implement TCP

bodiddles committed:
[RUNTIME] Enable TCP support

Adds TCP process and corresponding threads in Ansible.py.

TCP process creates TCP socket on port 1234 and starts two threads for sending and receiving
over the same socket.

Takes in commands from stateManager for handshake for studentCode upload and is able to send commands to Dawn.

Console logging is still untested

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral into runtime/TCPintegration

bodiddles committed:
tcp sorta works

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral into runtime/TCPintegration

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral into runtime/TCPintegration

bodiddles committed:
deleted sending student code error to dawn

brandonxxlee committed:
[RUNTIME] Add catch to student code error to send to dawn

brandonxxlee committed:
[RUNTIME] Create print method in student_code

bodiddles committed:
[RUNTIME] TCP integration

Adds TCP support for console logging and notifications with Dawn.

TCP process spawns once first message is received from Dawn and will send notifications and console logs via the notification proto over the tcp socekt.

Adds throttling to UDP receive process as well to hopefully improve performance.

Included compiled notification proto for now.

bodiddles committed:
Merge branch 'runtime/TCPintegration' of https://github.com/pioneers/PieCentral into runtime/student_printing

bodiddles committed:
removed unnecessary print statements in ansible

bodiddles committed:
[RUNTIME] Adds console logging proper

Overrides print for studentCode process and redirects to tcp socket to be sent to dawn.

Still maintains regular stdout print for testing purposes

bodiddles committed:
Fixed style issues and sleep logic for throttling

nikitakit reviewed (changes requested):
Could you please figure out why 11/11 tests are failing on travis?

bodiddles committed:
changed the print to stdout to print strings instead of tuples

nikitakit commented:
This mostly looks good, but I'll wait for TCP to be merged before reviewing it.

bodiddles committed:
changed stdout print to print *args rather than the joined string

yizow commented:
It looks like this shares a lot of commits with #138 TCP implementation
If that gets squashed and merged, will that cause merge conflicts here because of the squash?

bodiddles committed:
[Runtime] Changed throttling

Changed UDP Recv thread to not throttle.

Added code to TCP recv to better reflect its behavior.

bodiddles committed:
Removed UDP RECV_HZ and fixed whitespace

bodiddles committed:
minor text changes for udp ports

bodiddles committed:
Fixed more whitespace errors

bodiddles committed:
merged wth changes on TCP integration branch

brandonxxlee committed:
Merge branch 'master' into runtime/student_printing

nikitakit commented on runtime/testy/runtime.py:
```diff
@@ -139,9 +141,9 @@ def checkTimedOut(func, *args):
       testName += "_"
     setupFunc = getattr(studentCode, testName + "setup")
     mainFunc = getattr(studentCode, testName + "main")
-
```
This whitespace noise really clutters diffs. I think it's a good habit to run `git diff` before every commit, and undoing accidental whitespace changes before sending anything to github.

nikitakit commented on runtime/testy/studentAPI.py:
```diff
@@ -1,6 +1,7 @@
 import csv
 
 from runtimeUtil import *
+import builtins as __builtin__
```
No need to import builtins

nikitakit reviewed (changes requested):
Just a few minor style comments.

brandonxxlee committed:
[RUNTIME] Remove whitespace noise

brandonxxlee committed:
[RUNTIME] Remove unused import

nikitakit reviewed (approved):
lgtm

yizow reviewed (changes requested):
Test student code sending multiple items in one loop

brandonxxlee committed:
[RUNTIME] Increase throttle value

brandonxxlee committed:
[RUNTIME] On initial connect to Dawn send SensorMappings

bodiddles committed:
Fixed mapping code, compiled proto

Fixed csv parsing to read as not bytes, compiled new version of notification.proto, and updated fake_dawn to be able to test sending to sensor mapping.

nikitakit commented:
Changes still lgtm, though the throttling business is still slightly unsatisfying. I think we should consider adopting a more principled approach after first shipping v1 of a working integrated system.

That said, @yizow reminds me that there are no associated tests. Ideally, there would be some way of having travis CI be able to detect changes that break TCP functionality, so we don't discover this by way of shipping broken releases.

brandonxxlee committed:
[RUNTIME] Changed field names to be more accurate

brandonxxlee committed:
Merge branch 'runtime/send_mappings' of github.com:pioneers/PieCentral into runtime/send_mappings

brandonxxlee committed:
[RUNTIME] Change proto values

bodiddles committed:
compiled renamed protos

bodiddles committed:
added throttling back to fake_dawn

yizow commented on runtime/testy/Ansible.py:
```diff
@@ -236,7 +236,7 @@ def start(self):
 
 class TCPClass(AnsibleHandler):
-    TCP_HZ = 25
+    TCP_HZ = 180
```
Move to global constant

yizow commented on runtime/testy/runtime.py:
```diff
@@ -95,6 +95,8 @@ def nonTestModePrint(*args):
         nonTestModePrint(newBadThing.data)
         if newBadThing.event in restartEvents:
+          if newBadThing.event in studentErrorEvents:
+            stateQueue.put([SM_COMMANDS.SEND_CONSOLE, [newBadThing.stackTrace]])
```
I dont think this contains the actual error. (newBadThing.event)

you can just pass in newBadThing as an argument though, instead of newBadThing.stacktrace.
newBadThing objects have a built in print method that does this already i believe.

bodiddles replied:
I think the newBadThing.event contains the enum for the actual bad event that we are checking in this case.

Also, the send console command expects a string rather than an object and we are directly setting a field in the proto to the argument used. We aren't printing it so I don't think the str method of the object would be called.

yizow reviewed (changes requested):
Few minor changes

brandonxxlee committed:
[RUNTIME] Remove extra diff

brandonxxlee committed:
Merge branch 'master' into runtime/send_mappings

bodiddles committed:
Changed hz to global variables, changed sending errors using the send console command

bodiddles committed:
fixed merge conflicts

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral into runtime/student_printing

bodiddles committed:
changed merge conflict with master
